File: BigDaddyBackend/Domain/Entities/Order.cs
namespace BigDaddyBackend.Domain.Entities;

public class Order
{
    public int Id { get; set; }
    public int TableNumber { get; set; }
    public string? Observations { get; set; }
    public decimal Total { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public List<OrderItem> Items { get; set; } = new();
}

public class OrderItem
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public int Quantity { get; set; }
    public decimal Price { get; set; }
    public int OrderId { get; set; }
}

File: BigDaddyFrontend/assets/js/order.js
document.getElementById("finalizeOrder").addEventListener("click", async () => {
  const table = document.getElementById("tableNumber").value.trim();
  const obs = document.getElementById("obs").value.trim();

  if (!table || cart.length === 0) {
    alert("Preencha a mesa e adicione itens!");
    return;
  }

  const payload = {
    tableNumber: Number(table),
    observations: obs || null,
    items: cart.map(i => ({
      id: i.id || 0,
      name: i.name,
      quantity: i.quantity,
      price: Number(i.price)
    }))
  };

  console.log("Enviando para o backend:", payload); 

  try {
    const res = await fetch("http://localhost:5249/api/Order", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify(payload)
    });

    const texto = await res.text();
    if (res.ok) {
      alert("PEDIDO ENVIADO COM SUCESSO! ID: " + (JSON.parse(texto).orderId || "OK"));
      cart = [];
      updateCart();
      document.getElementById("tableNumber").value = "";
      document.getElementById("obs").value = "";
    } else {
      alert("ERRO " + res.status + "\n" + texto);
    }
  } catch (e) {
    alert("Backend não está rodando!");
  }
});